packages/syft/src/syft/service/user/user_service.py

```diff
@@ -342,8 +343,12 @@ def register(
             if new_user.created_by is None
             else self.get_role_for_credentials(new_user.created_by)
         )
-        can_user_register = context.node.metadata.signup_enabled or (
-            request_user_role in DATA_OWNER_ROLE_LEVEL
+        can_user_register = (
+            context.node.metadata.signup_enabled
```

Comment:
Move the check to `create_initial_settings`, check for node type there and set
```
if self.node_type == Enclave:
new_settings = NodeSettings(
                    name=self.name,
                    deployed_on=datetime.now().date().strftime("%m/%d/%Y"),
                    signup_enabled=True,
                )
```

Reply:
Resolved by [6d4112a](https://github.com/OpenMined/PySyft/commit/6d4112a96efdfb04cb7f26170afdf26624a74074)